### PR TITLE
Make the package noarch for Linux and Mac

### PR DESCRIPTION
**Pull request by ccordoba12**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

- This will reduce the number of packages we need to build for each version.
- It'll also help us to build our installers faster.
- Supersedes #177.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,18 +12,21 @@ source:
     - osx-zmq.patch
 
 build:
-  number: 1
-  skip: true  # [osx and arm64 and py < 39]
+  number: 2
   entry_points:
     - spyder = spyder.app.start:main
   osx_is_app: true
+  # https://github.com/conda/conda-build/issues/5385
```
Comment (hmaarrfk) [inline recipe/meta.yaml:19]: wow i forgot about this. glad i opened an issue for it......

**Comment by hmaarrfk** [comment]
I guess you need to weigh the reduction in the jobs against the added complexity and linter failures....

**Comment by ccordoba12** [comment]
> I guess you need to weigh the reduction in the jobs against the added complexity and linter failures....

I'm fine with the complexity (I don't think it makes things too different from what we have right now). But I was about to ask you if there's something we can do about the linter, i.e. is there something like `# noqa` to disable this specific warning?

**Comment by hmaarrfk** [comment]
i'm not too sure.

I'm half thinking if there is a way to maybe fix the offending character in the ico file...

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
hmm, seems like this is a real problem: https://github.com/conda-forge/spyder-feedstock/pull/192#issuecomment-2381450799

**Comment by ccordoba12** [comment]
So, is the linter failure a bug in conda-smithy?

**Comment by hmaarrfk** [comment]
No. It’s actually indicating that the new dependency necessitates an arch dependent package.

**Comment by ccordoba12** [comment]
Ok, I see. The problem is `fcitx-qt5`, which was added to help people from East Asian countries to type characters for their languages in Spyder.

A couple of additional questions for you:

* If we change the selector to `linux` to make the package `noarch`, then it won't be possible to install Spyder in aarch64, right?
* If that's the case, could you give us a hand to compile `fcitx-qt5` for that architecture? I know it's too much to ask, but we're really swamped with other things.

**Comment by hmaarrfk** [comment]
So i'm helping with PRs, but.....
Do you really want to have a hard dependency on a 2 year old package?

**Comment by hmaarrfk** [comment]
https://github.com/fcitx/fcitx-qt5

**Comment by hmaarrfk** [comment]
Ok with https://github.com/conda-forge/fcitx-qt5-feedstock/pull/5 merged soon you should be good to go.

**Comment by ccordoba12** [comment]
> Ok with https://github.com/conda-forge/fcitx-qt5-feedstock/pull/5 merged soon you should be good to go.

That's awesome! Thanks @hmaarrfk!

> Do you really want to have a hard dependency on a 2 year old package?

We received several requests about that. And I don't know about other Qt packages that provide input facilities on Linux for East Asian languages. So, I prefer to deal with that inconvenience rather than not giving those users a better user experience when working with Spyder.

**Comment by hmaarrfk** [comment]
understood!

**Comment by hmaarrfk** [comment]
> And I don't know about other Qt packages that provide input facilities on Linux for East Asian languages.

I always thought that this was Gnome's job, to provide some input language and that the editors were just receiving some unicode codepoint to display...

**Comment by ccordoba12** [comment]
> I always thought that this was Gnome's job, to provide some input language and that the editors were just receiving some unicode codepoint to display...

I think the problem is Conda envs are like chroots (not exactly virtual machines but like a Linux system within the host one). Since they come with their own set of Xorg/Qt/Gtk packages, they can't use the facilities provided by the host system.

**Comment by hmaarrfk** [comment]
>  Xorg/Qt/Gtk packages, they can't use the facilities provided by the host system.

this sounds like a real conda-forge problem. I don't know anybody that uses Pinyin type inputs on Ubuntu and I've never used it myself so I can't really test....

File: conda-forge.yml [hunk 2]
```diff
@@ -13,3 +13,7 @@ provider:
   linux_aarch64: default
   win: azure
 test: native_and_emulated
+noarch_platforms:
+  - linux_64
+  - osx_64
+  - win_64
```
Comment (ccordoba12) [inline conda-forge.yml:19]: @hmaarrfk, do we need `win_64` here or can we remove it for now?

Other than that, this should be ready to go, right?

Comment (hmaarrfk) [reply conda-forge.yml:19]: i think we need it, otherwise it won't trick conda-smithy to generate the windows builds. maybe. i'm not sure, try without it?

**Comment by ccordoba12** [comment]
@conda-forge-admin, please rerender

**Comment by ccordoba12** [comment]
Ok, I think this one is ready. Thanks @hmaarrfk for all your help with it!

**Comment by hmaarrfk** [comment]
Great. Hopefully this helps more than it causes churn

**Pull request merged**